docs/stdlib/constraints.rst

```diff
@@ -224,9 +224,8 @@ Constraints
             constraint exclusive on ( (.x, .y) );
         }
 
-    In principle, many possible expressions can appear in the ``on
-    (<expr>)`` clause of the ``exclusive`` constraint with a few
-    caveats:
+    Any possible expression can appear in the ``on (<expr>)`` clause of
+    the ``exclusive`` constraint as long as it adheres to the following:
 
     * The expression can only contain references to the immediate
```

Comment:
While we're in here, it would be really cool if we could show examples of these. For each item, I would love to see one example that works and one that doesn't. If that's too much, maybe we just explain these better. Areas of confusion for me:

- What does it mean that the expression can contain a reference to "links of the type?" I thought constraints could not dig into links. (I think it may mean you can constrain on the link itself but not on properties of the linked object.)
- What is a "long path?"
- How do I know which functions are `Immutable`?

Reply:
Actually, this probably isn't in scope for this change. I'll file a new issue for it.

Reply:
https://github.com/edgedb/edgedb/issues/6217